src/odemis/acq/stream_settings.py

```diff
@@ -0,0 +1,174 @@
+        """
+        :param file_path: full directory path to save the JSON file
+        :param max_entries: number of entries to save in the JSON file
+        :param config_data: list of stream controller settings where the first element is the latest
+         stream entry. The number of entries is defined by max_entries.
+        """
+        self.max_entries = max_entries
+        self.config_data = config_data
+        self.file_path = file_path
+        self._read()  # update the config_data by reading the JSON file
+        self.settings_order = None  # the ordered list of setting parameters used for loading any given stream
+
+    def _read(self):
+        logging.debug("Reading the most recently used stream settings in JSON")
+        if os.path.exists(self.file_path):
+            try:
+                with open(self.file_path, 'r') as jsonfile:
+                    self.config_data = json.load(jsonfile)
+            except json.decoder.JSONDecodeError as e:
+                logging.warning(f"Error decoding JSON: {e}")
+            except Exception as e:
+                logging.warning(f"Error in reading the stream settings: {e}")
+
+    def _write(self):
+        logging.debug("Writing the most recently used stream settings in JSON")
+        try:
+            if self.config_data:
+                with open(self.file_path, 'w') as jsonfile:
+                    json.dump(self.config_data, jsonfile, indent=2)
+        except Exception as e:
+            logging.warning(f"Error in writing the stream settings: {e}")
+
+    def update_data(self, new_entries: List[Dict[str, Any]]):
+        """
+        Update the saved stream setting with the list of new_entries. First element in JSON file
+        is the latest entry while the last element is the oldest stream setting. The file updates
+        its entries based on the new_entries while maintaining a fixed length of entries in the file.
+        :param new_entries: (list of stream controller settings) Each element in the list represent the setting
+         of the stream saved as a list from oldest to latest stream setting
+        """
+        # Convert config data to list to enable list operations
+        if self.config_data is None:
+            self.config_data = []
+        data = self.config_data
+
+        for e in new_entries:
+            # find if the new entry name exists in the saved data
+            index = get_config_index(data, e["name"])
+            # pop the entry from the list if the stream name is same to avoid duplication
+            if index is not None:
+                data.pop(index)
+            # insert the new entry at the first position
+            data.insert(0, e)
+            # length of list of entries should be constant
+            data = data[:self.max_entries]
+
+        self.config_data = data
+        self._write()
+
+    def get_settings_order(self, stream: FluoStream):
```

Comment:
A "get" function has to return something. That's the convention. I think here you're doing some sort of caching, which should really not happen anyway. => just always run the code, and return the order.

Also, it would then become a simple function, which can be moved outside of this class.